src/utils/use-media-query.ts

```diff
@@ -1,45 +1,27 @@
-import { useState, useEffect } from 'react';
+import { useState, useCallback, useEffect } from 'react';
 
-function hasWindow() {
-  return typeof window !== 'undefined';
-}
+export const useMediaQuery = (query: string) => {
+  const [targetReached, setTargetReached] = useState(false);
 
-export const useMediaQuery = (mediaQuery: string) => {
-  const [isMatched, setMatched] = useState(() => {
-    if (!hasWindow()) return false;
-    return Boolean(window.matchMedia(mediaQuery).matches);
-  });
+  const updateTarget = useCallback((e: MediaQueryListEvent) => {
+    if (e.matches) {
+      setTargetReached(true);
+    } else {
+      setTargetReached(false);
+    }
+  }, []);
 
   useEffect(() => {
-    if (!hasWindow()) return;
-    const mediaQueryList = window.matchMedia(mediaQuery);
-    const documentChangeHandler = () =>
-      setMatched(Boolean(mediaQueryList.matches));
-    listenTo(mediaQueryList, documentChangeHandler);
+    const media = window.matchMedia(query);
+    media.addEventListener('change', updateTarget);
```

Comment:
Safari did/didn't support addEventListener on MediaQueryList due to an overly strict following of [the spec](https://drafts.csswg.org/cssom-view/#the-mediaquerylist-interface). Have you verified that this is no longer the case in most used Safari versions?

Reply:
Yes, seems to be updated in Safari 14
<img width="605" alt="Screenshot 2022-11-06 at 20 01 58" src="https://user-images.githubusercontent.com/606374/200189895-16d6dc66-9962-4d75-b7ce-03bfe91c4a42.png">
<img width="430" alt="Screenshot 2022-11-06 at 20 00 33" src="https://user-images.githubusercontent.com/606374/200189897-311e051e-3114-4657-b02c-7c0633c056cd.png">